docs/public/components/testing.md

```diff
@@ -0,0 +1,203 @@
+
+=== ":octicons-file-code-16: comp/compression/impl-zstd/component_test.go"
+    ```go
+    package implzstd
+    
+    import (
+      "testing"
+
+      configmock "github.com/DataDog/datadog-agent/comp/core/config/mock"
+      logmock "github.com/DataDog/datadog-agent/comp/core/log/mock"
+    )
+    
+    func TestCompress(t *testing.T) {
+      logComponent := configmock.New(t)
+      configComponent := logmock.New(t)
+      
+      requires := Requires{
+        Conf: configComponent,
+        Log: logComponent,
+      }
+      
+      provides := NewComponent(requires)
+      component := provides.Comp
+      
+      result, err := component.Compress([]byte("Hello World"))
+      assert.Nil(t, err)
+  
+      assert.Equal(t, ..., result)
+    }
+    ```
+
+### Testing lifecycle hooks
+
+Sometimes a component uses [Fx lifecycle](fx.md#lifecycle) to add hooks. It is a good practice to test the hooks as well. 
+
+For this example, imagine a component wants to add some hooks into the app lifecycle. Some code is omitted for simplicity:
+
+=== ":octicons-file-code-16: comp/somecomponent/impl/component.go"
+    ```go
+    package impl
+
+    import (
+      "context"
+      
+      somecomponent "github.com/DataDog/datadog-agent/comp/somecomponent/def"
+      compdef "github.com/DataDog/datadog-agent/comp/def"
+    )
+
+    type Requires struct {
+      Lc      compdef.Lifecycle
+    }
+
+    type Provides struct {
+      Comp somecomponent.Component
+    }
+
+    type component struct {
+      started  bool
+      stopped bool
+    }
+
+    func (c *component) start() error {
+      // [...]
+      
+      c.started = true
+
+      return nil
+    }
+
+    func (h *healthprobe) stop() error {
+      // [...]
+      
+      c.stopped = true
+      c.started = false
+
+      return nil
+    }
+
+    // NewComponent creates a new healthprobe component
+    func NewComponent(reqs Requires) (Provides, error) {
+      provides := Provides{}
+      comp := &component{}
+
+      reqs.Lc.Append(compdef.Hook{
+        OnStart: func(ctx context.Context) error {
+          return comp.start()
+        },
+        OnStop: func(ctx context.Context) error {
+          return comp.stop()
+        },
+      })
+
+      provides.Comp = comp
+      return provides, nil
+    }
+    ```
+
+The goal is to test that the component updates the `started` and `stopped` fields.
+
+To accomplish this, create a new lifecycle instance, create a `Require` struct instance, initialize the component, and validate that calling `Start` on the lifecycle instance calls the component hook and executes the logic.
+
+To create a lifecycle instance, use the helper function `compdef.NewTestLifecycle()`. The function returns a lifecycle wrapper that can be used to populate the `Requires` struct. The `Start` and `Stop` functions can also be called.
+
+!!! Info 
+    <!-- TODO add link to NewTestLifecycle function once this PR https://github.com/DataDog/datadog-agent/pull/25184 is merged-->
```

Comment:
This has been merged.

Link to function: https://github.com/DataDog/datadog-agent/blob/649508917c80b804e150e0d61a4ff7564afb7852/comp/def/lifecycle_mock.go#L21